internal/metricstestutils/metrics_testutils.go

```diff
@@ -0,0 +1,159 @@
+	return metric(metricspb.MetricDescriptor_CUMULATIVE_DISTRIBUTION, name, keys, timeseries)
+}
+
+// Summary creates a summary metric.
+func Summary(name string, keys []string, timeseries ...*metricspb.TimeSeries) *metricspb.Metric {
+	return metric(metricspb.MetricDescriptor_SUMMARY, name, keys, timeseries)
+}
+
+// Timeseries creates a timeseries. It takes the start time stamp, a sequence of label values (associated
+// with the label keys in the overall metric), and the value of the timeseries.
+func Timeseries(sts time.Time, vals []string, point *metricspb.Point) *metricspb.TimeSeries {
+	return &metricspb.TimeSeries{
+		StartTimestamp: ToTS(sts),
+		Points:         []*metricspb.Point{point},
+		LabelValues:    toVals(vals),
+	}
+}
+
+// Double creates a double point.
+func Double(ts time.Time, value float64) *metricspb.Point {
+	return &metricspb.Point{Timestamp: ToTS(ts), Value: &metricspb.Point_DoubleValue{DoubleValue: value}}
+}
+
+// Dist creates a distribution point. It takes the time stamp, the bucket boundaries for the distribution, and
+// the and counts for the individual buckets as input.
+func Dist(ts time.Time, bounds []float64, counts []int64) *metricspb.Point {
+	var count int64
+	var sum float64
+	buckets := make([]*metricspb.DistributionValue_Bucket, len(counts))
+
+	for i, bcount := range counts {
+		count += bcount
+		buckets[i] = &metricspb.DistributionValue_Bucket{Count: bcount}
+		// create a sum based on lower bucket bounds
+		// e.g. for bounds = {0.1, 0.2, 0.4} and counts = {2, 3, 7, 9)
+		// sum = 0*2 + 0.1*3 + 0.2*7 + 0.4*9
+		if i > 0 {
+			sum += float64(bcount) * bounds[i-1]
+		}
+	}
+	distrValue := &metricspb.DistributionValue{
+		BucketOptions: &metricspb.DistributionValue_BucketOptions{
+			Type: &metricspb.DistributionValue_BucketOptions_Explicit_{
+				Explicit: &metricspb.DistributionValue_BucketOptions_Explicit{
+					Bounds: bounds,
+				},
+			},
+		},
+		Count:   count,
+		Sum:     sum,
+		Buckets: buckets,
+		// There's no way to compute SumOfSquaredDeviation from prometheus data
+	}
+	return &metricspb.Point{Timestamp: ToTS(ts), Value: &metricspb.Point_DistributionValue{DistributionValue: distrValue}}
+}
+
+// Summ creates a summary point.
+func Summ(ts time.Time, count int64, sum float64, percent, vals []float64) *metricspb.Point {
+	percentiles := make([]*metricspb.SummaryValue_Snapshot_ValueAtPercentile, len(percent))
+	for i := 0; i < len(percent); i++ {
+		percentiles[i] = &metricspb.SummaryValue_Snapshot_ValueAtPercentile{Percentile: percent[i], Value: vals[i]}
+	}
+	summaryValue := &metricspb.SummaryValue{
+		Sum:   &wrapperspb.DoubleValue{Value: sum},
+		Count: &wrapperspb.Int64Value{Value: count},
+		Snapshot: &metricspb.SummaryValue_Snapshot{
+			PercentileValues: percentiles,
+		},
+	}
+	return &metricspb.Point{Timestamp: ToTS(ts), Value: &metricspb.Point_SummaryValue{SummaryValue: summaryValue}}
+}
+
+// ToTS creates a timestamp.
+func ToTS(ts time.Time) *timestamppb.Timestamp {
```

Comment:
nit: Should this be simply Timestamp instead fo ToTS? ToTS seems to suggest that it is converting time to Timestamp not creating Timestamp.

Reply:
Done.